src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -8,6 +8,7 @@
 using System.Linq;
```

Comment:
After the change it feels like the `WorkspaceContext` code become a little more messy and clumsy. It is understandable since it now suites more usages.  I'm wondering is it possibly this class can be broken down further to produce a cleaner solution. For example the function `GetRuntimeContext`, can it be moved to a utility class or a helper. And the `WorkspaceContext` can be really a place simply to cache objects other that filter and extract contexts suits different conditions.